### PR TITLE
Clim core rework

### PR DESCRIPTION
**Pull request by dkochmanski**
This PR doesn't introduce any new functionality, it moves the code around to be easier to navigate. Presentations gain a new module and different theme abstractions are in a single file. Most of the `accept` implementation is put in the input-editing.lisp file -- that shows that this file needs some work with an eye on the spec to figure out, what belongs where (i.e complete-input).

### CODE REVIEW

File: Core/clim-core/presentation-defs.lisp [hunk 1]
```diff
@@ -1304,11 +820,6 @@ protocol retrieving gestures from a provided string."))
   (declare (ignore acceptably for-context-type))
   (prin1 object stream))
 
-(defmethod presentation-type-of ((object symbol))
-  (if (eq (symbol-package object) (find-package :keyword))
-      'keyword
-      'symbol))
```
Comment (scymtym) [inline Core/clim-core/presentation-defs.lisp:1310]: Why are these moved? I think it makes sense to defined standard presentation types along with "their" methods.

File: Core/clim-basic/decls.lisp [hunk 2]
```diff
@@ -988,6 +992,8 @@ panes."))
     (frame input-context stream x y &key event))
 (defgeneric frame-input-context-button-press-handler
     (frame stream button-press-event))
+(defgeneric frame-input-context-track-pointer
+    (frame input-context stream event))
```
Comment (scymtym) [inline Core/clim-basic/decls.lisp:996]: Maybe add a comment to the effect that this is a non-standard extension.

File: Core/clim-core/presentations/typed-output.lisp [hunk 3]
```diff
@@ -48,3 +48,156 @@
...
+  (type-key parameters options object type stream view
+   &key &allow-other-keys))
+
+(define-default-presentation-method present
+    (object type stream (view textual-view) &key acceptably for-context-type)
+  (declare (ignore for-context-type type))
+  (if acceptably
+      (let ((*print-readably* t))
+        (prin1 object stream))
+      (princ object stream)))
```
Comment (scymtym) [inline Core/clim-core/presentations/typed-output.lisp:65]: There is another definition of this method in `standard-presentations.lisp` (or may `presentation-defs.lisp` in this commit).

File: Core/clim-core/presentations/presentation-type-functions.lisp [hunk 4]
```diff
@@ -0,0 +1,399 @@
...
+    (when (and (typep name '(or symbol class))
+               (get-ptype-metaclass name))
+      (funcall-presentation-generic-function presentation-type-specifier-p object))))
+
+(define-presentation-generic-function %presentation-typep presentation-typep
+  (type-key parameters object type))
+
+(define-default-presentation-method presentation-typep (object type)
+  (declare (ignore object type))
+  nil)
```
Comment (scymtym) [inline Core/clim-core/presentations/presentation-type-functions.lisp:90]: This seems to conflict with `(define-presentation-method presentation-typep (object (type t)) …)` in `presentation-defs.lisp`.

File: Core/clim-basic/decls.lisp (hunk 2 above)
Comment (dkochmanski) [reply Core/clim-basic/decls.lisp:996]: I think that this is implicit from the clim package, from the clim-internals perspective it doesn't matter

File: Core/clim-core/presentations/typed-output.lisp (hunk 3 above)
Comment (dkochmanski) [reply Core/clim-core/presentations/typed-output.lisp:65]: define-default-presentation-method and define-presentation-method are two very different macros, even expansion differ. I wouldn't touch it for now since this PR is about rearranging things.

File: Core/clim-core/presentations/presentation-type-functions.lisp (hunk 4 above)
Comment (dkochmanski) [reply Core/clim-core/presentations/presentation-type-functions.lisp:90]: ditto, default method seems to do the same, but without decoding types.

File: Core/clim-core/presentation-defs.lisp (hunk 1 above)
Comment (dkochmanski) [reply Core/clim-core/presentation-defs.lisp:1310]: presentation-type-of is specified as a function in this part of the spec. moreover, it doesn't return the presentation-type, only the symbol. I think that it is justified to put it there.

File: Core/clim-core/presentations/presentation-type-functions.lisp [hunk 5]
```diff
@@ -0,0 +1,399 @@
+;;; ---------------------------------------------------------------------------
+;;;     Title: Typed output
```
[the remark this answers is not in the document]
Comment (dkochmanski) [reply Core/clim-core/presentations/presentation-type-functions.lisp:2]: fixed

File: Core/clim-core/theming.lisp [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+;;;  (c) copyright 2001 by Michael McDonald <mikemac@mikemac.com>
+;;;  (c) copyright 2001 by Gilbert Baumann <unk6@rz.uni-karlsruhe.de>
+;;;  (c) copyright 2020 by Daniel Kochmański <daniel@turtleware.eu>
+;;;
+;;; ---------------------------------------------------------------------------
+;;;
+;;; Theming is a frequently requested feature. For now we'll just move
+;;; things related to theming to a single file so it is easier to come
+;;; up with the unified abstraction later on.
+;;;
```
[the remark this answers is not in the document]
Comment (dkochmanski) [reply Core/clim-core/theming.lisp:17]: This follows the convention estabilished by Gilberth in soem files. IMO it makes sense to show, when the file was created and by whom. As of the title and notes below copyrights, it is also justified, because it is the best place to put todos (and in clim source code it is often done so), so it is the commiter responsibility to update comments. getting out of sync is of course possible, but I'd see that rather as an issue.

File: Core/clim-core/clim-core.asd [hunk 7]
```diff
@@ -24,17 +25,10 @@
...
-                               "input-editing" "frames"))
-   (:file "dialog" :depends-on ("panes" "frames" "incremental-redisplay"
-                                "table-formatting" "presentations"
-                                "bordered-output" "presentation-defs"
-                                "input-editing" "commands" "gadgets"))
-   (:file "builtin-commands" :depends-on ("table-formatting" "commands" "presentations"
-                                          "dialog" "presentation-defs" "input-editing"))))
+   (:file "menu-choose" :depends-on ("commands" "table-formatting" "presentation-defs" "panes" "frames" "presentations"))
+   (:file "panes" :depends-on ("incremental-redisplay" "presentations" "presentation-defs" "input-editing" "frames"))
+   (:file "dialog" :depends-on ("panes" "frames" "incremental-redisplay" "table-formatting" "presentations" "bordered-output" "presentation-defs" "input-editing" "commands" "gadgets"))
```
[the remark this answers is not in the document]
Comment (dkochmanski) [reply Core/clim-core/clim-core.asd]: I think that for asd files it only introduces noise. I don't think we have it documented anywhere.

File: Core/clim-core/clim-core.asd [hunk 8]
```diff
@@ -4,17 +4,18 @@
...
   :components
   ((:file "defresource")
-   (:file "presentations")
-   (:file "presentation-translators" :depends-on ("presentations"))
-   (:file "presentation-drag-and-drop" :depends-on ("presentations" "presentation-translators"))
+   (:module "presentations"
+    :serial t
+    :components ((:file "presentations")
+                 (:file "presentation-translators")
+                 (:file "presentation-drag-and-drop")))
```
[the remark this answers is not in the document]
Comment (dkochmanski) [reply Core/clim-core/clim-core.asd]: yes,for these two files it makes sense, pushed.

File: Core/clim-core/clim-core.asd [hunk 9]
```diff
@@ -4,17 +4,18 @@
   :depends-on ("clim-basic" #+sbcl (:require "sb-introspect"))
```
[the remark this answers is not in the document]
Comment (dkochmanski) [reply Core/clim-core/clim-core.asd:4]: fixed

File: Core/clim-core/clim-core.asd [hunk 10]
```diff
@@ -4,20 +4,21 @@
...
```
[the remark this answers is not in the document]
Comment (dkochmanski) [reply Core/clim-core/clim-core.asd:4]: gadget-view-* is defined with views in clim-basic, I don't know if it is referenced or not. I've added to the commit note about referencing #842.

File: Core/clim-core/clim-core.asd [hunk 11]
```diff
@@ -7,8 +7,10 @@
    (:file "theming")
```
[the remark this answers is not in the document]
Comment (dkochmanski) [reply Core/clim-core/clim-core.asd:7]: fixed
